src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java

```diff
@@ -722,7 +734,11 @@ public SCMRevision run(GitClient client, String remoteName) throws IOException,
             }
             Set<String> result = new HashSet<>();
             for (SCMRevision r : telescope.getRevisions(remote, credentials, referenceTypes)) {
-                result.add(r.getHead().getName());
+                if (r instanceof GitTagSCMRevision && context.wantTags()) {
```

Comment:
Was not following the same contract as without a telescope, found thanks to test